### PR TITLE
Enable strict concurrency checks on `Basics` module

### PR DESCRIPTION
**Pull request by MaxDesiatov**
Also refactored `URLSessionHTTPClient` to remove `weak` references, which were incompatible with strict concurrency checks.

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Review by xedin — APPROVED** [review]

File: Package.swift [hunk 1]
```diff
@@ -193,7 +192,10 @@ let package = Package(
                 .product(name: "SwiftToolsSupport-auto", package: "swift-tools-support-core"),
                 .product(name: "SystemPackage", package: "swift-system"),
             ],
-            exclude: ["CMakeLists.txt", "Vendor/README.md"]
+            exclude: ["CMakeLists.txt", "Vendor/README.md"],
+            swiftSettings: [
+                .enableExperimentalFeature("StrictConcurrency=complete"),
```
[the remark this answers is not in the document]
Comment (MaxDesiatov) [reply Package.swift:197]: Would you prefer to just use `StrictConcurrency` here? I'm also not sure if `StrictConcurrency` is equivalent to all other options here in Swift 5.8/5.9, which is what we use on CI.

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

File: Package.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (xedin) [reply Package.swift:197]: Yeah, `StrictConcurrency` without modifier defaults to `complete` in 5.9 on

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Pull request merged**